dynawo/sources/Common/test/TestValidateDic.cpp

```diff
@@ -28,22 +28,77 @@ TEST(Models, TestBuildCheckSum) {
+    "Error: File: dic/noSeparator/NoSeparator.dic "
+    "line: 'MyLabel  Separator is missing !' is not well defined, no separator '=' between the key and the value.");
+  ssPython.str(std::string());
+
+  cmd = pythonCmd + " validateDictionaries.py --inputDir dic/capitalInLog/";
+  executeCommand(cmd, ssPython);
+  result = ssPython.str();
+  boost::erase_all(result, "\n");
+  ASSERT_EQ(result, "Executing command : " + cmd +
+    "Error: File: dic/capitalInLog/CapitalInLog.dic "
+    "line: 'MyLabel = First word begins with a capital letter !', the value definition should not begin with a capital letter.");
+  ssPython.str(std::string());
+
+  cmd = pythonCmd + " validateDictionaries.py --inputDir dic/differentKeys/";
+  executeCommand(cmd, ssPython);
+  result = ssPython.str();
+  boost::erase_all(result, "\n");
+  ASSERT_EQ(result, "Executing command : " + cmd +
+    "Error: ['dic/differentKeys/DifferentKeys_en_GB.dic'] and "
+    "['dic/differentKeys/DifferentKeys_fr_FR.dic'] have not the same entries.");
+  ssPython.str(std::string());
+
+  cmd = pythonCmd + " validateDictionaries.py --inputDir dic/argsCount/";
+  executeCommand(cmd, ssPython);
+  result = ssPython.str();
+  boost::erase_all(result, "\n");
+  ASSERT_EQ(result, "Executing command : " + cmd +
+    "Error: Messages of key 'Hello' for dictionary ['dic/argsCount/ArgsCount_en_GB.dic'] and "
+    "['dic/argsCount/ArgsCount_fr_FR.dic'] have not the same number of arguments.");
+  ssPython.str(std::string());
+
+  cmd = pythonCmd + " validateDictionaries.py --inputDir dic/timelinePriorityLocale/";
+  executeCommand(cmd, ssPython);
+  result = ssPython.str();
+  boost::erase_all(result, "\n");
+  ASSERT_EQ(result, "Executing command : " + cmd +
+    "Error: MyTimelinePriority dictionary should be present only once and without a locale.");
+  ssPython.str(std::string());
+
+  cmd = pythonCmd + " validateDictionaries.py --inputDir dic/noTimeline/";
+  executeCommand(cmd, ssPython);
+  result = ssPython.str();
+  boost::erase_all(result, "\n");
+  ASSERT_EQ(result, "Executing command : " + cmd +
+    "Error: MyTimeline dictionary not present in input directories while "
+    "['dic/noTimeline/MyTimelinePriority.dic'] is.");
+  ssPython.str(std::string());
+
+  cmd = pythonCmd + " validateDictionaries.py --inputDir dic/ --modelicaDir dic/";
+  executeCommand(cmd, ssPython);
+  result = ssPython.str();
+  boost::erase_all(result, "\n");
+  ASSERT_EQ(result, "Executing command : " + cmd + "Usage: validateDictionaries.py "
+    "--inputDir=<directories> [--outputDir=<directory> [--namespace=<namespace>] [--existingKeysDir=<directories>]] "
+    "[--modelicaDir=<directory> --modelicaPackage=<packageName>]"
+    "    Script checks integrity of all dictionaries in inputDir."
+    "    If check is ok, it generates keys.h and keys.cpp files in outputDir and keys.mo files in modelicaDir."
+    "validateDictionaries.py: error: Parent package of modelica keys files should be informed.");
```

Comment:
Can we add a "full" test that works with multiple locales and a priority file?

Reply:
Add an all in one test case